### PR TITLE
Fix regenerate option

### PR DESCRIPTION
**Pull request by quisse**
Fixing #207

### CODE REVIEW

File: src/Doctrine/EntityRegenerator.php [hunk 1]
```diff
@@ -260,8 +260,8 @@ private function getMappedFieldsInEntity(ClassMetadata $classMetadata)
 
             // exclude inherited properties
             $targetFields = array_filter($targetFields, function ($field) use ($classReflection) {
-                return $classReflection->hasProperty($field) &&
-                    $classReflection->getProperty($field)->getDeclaringClass()->getName() == $classReflection->getName();
+                return ($classReflection->hasProperty($field) &&
+                        $classReflection->getProperty($field)->getDeclaringClass()->getName() == $classReflection->getName()) || !$classReflection->hasProperty($field);
```
Comment (weaverryan) [inline src/Doctrine/EntityRegenerator.php]: Can you explain what case this is catching exactly?

File: src/Doctrine/DoctrineHelper.php [hunk 2]
```diff
@@ -161,16 +161,32 @@ public function getMetadata(string $classOrNamespace = null, bool $disconnected
...
+                    if (null === $classOrNamespace) {
                         $metadata[$m->getName()] = $m;
+                    } else {
+                        if ($m->getName() === $classOrNamespace) {
+                            return $m;
+                        }
+
+                        if (0 === strpos($m->getName(), $classOrNamespace)) {
+                            $metadata[$m->getName()] = $m;
+                        }
```
Comment (weaverryan) [inline src/Doctrine/DoctrineHelper.php]: So, we wrap everything in a catch for `ReflectionException`. Can you explain what's happening here? What does that exception say?

**Comment by quisse** [comment]
Hi @weaverryan, I'm quite busy at the moment but will get into it anytime soon. To be continued...

**Comment by weaverryan** [comment]
Friendly ping!

**Comment by quisse** [comment]
Thanks for the ping. I'm still quite busy, will try to look into this later. My solution isn't working 100% I've already noticed but it helps enough for now. It's indeed super technical and we needed a quick fix. If someone who's more at home in this, wants to take it up that would also be fine for me.

**Comment by quisse** [comment]
Due to lack of time I'll have to close this PR. My apologies & thanks for your time!

**Pull request closed without merging**